sequencer/src/bin/nasty-client.rs

```diff
@@ -940,11 +936,13 @@ impl ResourceManager<BlockQueryData<SeqTypes>> {
                     .context(format!("fetching header {block}"))
             })
             .await?;
-        if header.ns_table.is_empty() {
+        let num_namespaces = header.ns_table.iter().count();
```

Comment:
Can't use `ExactSizeIterator` here because the only way to compute the length is by consuming the iterator because we currently allow duplicate namespace IDs in the namespace table and so we don't know which entries to ignore unless we scan the whole thing. I could add a `len()` method but it would just wrap `iter().count()`.

I think I'll leave it as-is for now. In the future we intend to enforce uniqueness of namespace IDs at the consensus level. When that happens we can deduce the number of entries in the namespace table in constant time and so a `len()` method would make more sense.